src/envs/stick_point.py

```diff
@@ -0,0 +1,475 @@
+                y = rng.uniform(self.rz_y_lb + radius, self.rz_y_ub - radius)
+                geom = utils.Circle(x, y, radius)
+                # Keep only if no intersections with existing objects.
+                if not any(geom.intersects(g) for g in collision_geoms):
+                    break
+            collision_geoms.add(geom)
+            theta = rng.uniform(self.theta_lb, self.theta_ub)
+            state_dict[self._robot] = {"x": x, "y": y, "theta": theta}
+            # Finally, sample the stick, making sure that the origin is in the
+            # reachable zone, and that it doesn't collide with anything.
+            radius = self.robot_radius + self.init_padding
+            while True:
+                # The radius here is to prevent the stick from being very
+                # slightly in the reachable zone, but not grabbable.
+                x = rng.uniform(self.rz_x_lb + radius, self.rz_x_ub - radius)
+                y = rng.uniform(self.rz_y_lb + radius, self.rz_y_ub - radius)
+                theta = rng.uniform(self.theta_lb, self.theta_ub)
+                rect = utils.Rectangle(x, y, self.stick_width,
+                                       self.stick_height, theta)
+                # Keep only if no intersections with existing objects.
+                if not any(rect.intersects(g) for g in collision_geoms):
+                    break
+            state_dict[self._stick] = {
+                "x": x,
+                "y": y,
+                "theta": theta,
+                "held": 0.0
+            }
+            init_state = utils.create_state_from_dict(state_dict)
+            task = Task(init_state, goal)
+            tasks.append(task)
+        return tasks
+
+    def _object_to_geom(self, obj: Object, state: State) -> _Geom2D:
+        x = state.get(obj, "x")
+        y = state.get(obj, "y")
+        if obj.is_instance(self._robot_type):
+            return utils.Circle(x, y, self.robot_radius)
+        if obj.is_instance(self._point_type):
+            return utils.Circle(x, y, self.point_radius)
+        assert obj.is_instance(self._stick_type)
+        theta = state.get(obj, "theta")
+        return utils.Rectangle(x=x,
+                               y=y,
+                               width=self.stick_width,
+                               height=self.stick_height,
+                               theta=theta)
+
+    def _stick_rect_to_tip_rect(
+            self, stick_rect: utils.Rectangle) -> utils.Rectangle:
+        theta = stick_rect.theta
+        width = self.stick_tip_width
+        scale = stick_rect.width - width
+        return utils.Rectangle(x=stick_rect.x + scale * np.cos(theta),
+                               y=stick_rect.y + scale * np.sin(theta),
+                               width=self.stick_tip_width,
+                               height=stick_rect.height,
+                               theta=theta)
+
+    def _get_grasp_point(self, state: State, stick: Object,
```

Comment:
rename this, "point" has a different meaning in this env, and also should include the word "stick": `_get_stick_grasp_loc`?